### PR TITLE
Update release scripts

### PR DESCRIPTION
**Pull request by igloosi**
This PR address the current long-winded release process, by merging a couple of steps.
Changes are:
- addition of two scripts: `npm run prerelease` and `npm run release`
-  addition of `pre-release.sh` shell script that runs:
    - `npm run build:packages`
    - `npm run build:dist`
    - `lerna publish --skip-git --skip-npm`
    and if all successful it
    - adds, commits and pushes all files to remote
- addition of `release.sh` shell script that runs:
    - `npm-publish.sh` shell script ffor each package
    - checks to see if you're logged into NPm as the correct user
    - creates and pushes the tag version to remote
    - create a zip of `dist` folder ready for attaching to release tab in Github
- addition of `npm-publish.sh` shell scripts that:
    - checks for each package that current version is not the same as the published version
    - if not runs `npm publish` for each package
- updates publish documentation
- adds change to CHANGELOG

Perhaps we could merge this into #545 and test with that as the documentation describes that package folder structure

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -17,9 +17,11 @@
     "preinstall": "node bin/check-nvmrc.js",
     "prestart": "node bin/check-nvmrc.js",
     "start": "gulp dev --destination 'public'",
+    "pre:release": "node bin/check-nvmrc.js && ./bin/pre-release.sh",
```
Comment (nickcolley) [inline package.json]: nit: prerelease (consistent with npm above)

File: bin/release.sh [hunk 2]
```diff
@@ -0,0 +1,38 @@
+#!/bin/sh
+set -e
+
+# echo  "Checking that you can publish to npm..."
+
+# at some point we should create a team and check if user exists in a team
+# ! npm team ls developers | grep -q $NPM_USER
+
+NPM_USER=$(npm whoami)
+if ! [[ "govuk-patterns-and-tools" = $NPM_USER ]]; then
```
Comment (nickcolley) [inline bin/release.sh]: 👏

File: bin/release.sh [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+if [[ $LATEST_PUBLISHED_TAG = $TAG ]]; then
+    echo ⚠️ Tag $TAG already exists
+    exit 1
+else
+    echo "🗒 Tagging repo using tag version: $TAG ..."
+    git tag $TAG -m "GOV.UK Frontend release $TAG"
+    git push --tags
+    echo "🗒 Tag $TAG created and push to remote."
+
+    echo "🗒 Creating a relase artifact..."
```
Comment (nickcolley) [inline bin/release.sh]: nit: relase -> release

**Comment by igloosi** [comment]
yes it will, but didn't want to update docs just yet, if we change things

File: bin/pre-release.sh [hunk 4]
```diff
@@ -0,0 +1,8 @@
+#!/bin/sh
+set -e
+
+ALL_PACKAGE_VERSION=$(node -p "require('./packages/all/package.json').version")
+TAG="v$ALL_PACKAGE_VERSION"
+
+npm run build:packages && npm run build:dist && lerna publish --skip-git --skip-npm && git add . && git commit -m "Release $TAG"
+echo "🗒 All done. Ready to create a pull request. Once approved, run npm run release"
```
Comment (alex-ju) [inline bin/pre-release.sh]: This message show us even when we have an error or we cancel Lerna. I'll give it a thought, maybe we'll be able to wrap it somehow.
screen shot 2018-02-26 at 12 55 11

Comment (igloosi) [reply bin/pre-release.sh]: ok, let's have chat how to best deal with it

Comment (igloosi) [reply bin/pre-release.sh]: @alex-ju have pushed an update to prerelease script. how about now?

Comment (igloosi) [reply bin/pre-release.sh]: if you cancel lerna publish, exit code is 0

Comment (alex-ju) [reply bin/pre-release.sh]: yep, looks good now.

**Comment by igloosi** [comment]
this also needs update release documentation and changelog

**Pull request merged**